infrastructure/iam.tf

```diff
@@ -53,27 +53,26 @@ module "iam_github_oidc_provider" {
 
 module "iam-assumable-role-with-oidc" {
   source  = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
-  version = "5.39.1"
   create_role = true
   role_name = "Ognjen-github-actions-role"
   provider_url = module.iam_github_oidc_provider.url
 
 
-  oidc_subjects_with_wildcards = ["repo:OgnjenDacevic/docker-nodejs-sample*"]
+  oidc_subjects_with_wildcards = ["repo:OgnjenDacevic/docker-nodejs-sample:*"]
   role_policy_arns = [module.iam_policy.arn]
+  oidc_fully_qualified_audiences = ["sts.amazonaws.com"]
 }
 
-module "iam-role-for-service-accounts-eks" {
+module "iam_role_for_service_accounts_eks" {
   source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
-  version = "5.39.1"
   role_name = "Ognjen-role-for-load-balancer-controller"
 
   attach_load_balancer_controller_policy = true
-
+  
   oidc_providers = {
     github = {
       provider_arn = module.eks.oidc_provider_arn
-      namespace_service_accounts = ["load-balancer-service-account:load-balancer-controller"]
+      namespace_service_accounts = ["load-balancer-service-account:${var.service_account_name}"]
```

Comment:
name namespace more meaningful, not to have service account in name as it has nothing to do with it.